packages/popover2/src/popover2.tsx

```diff
@@ -356,17 +360,26 @@ export class Popover2<T> extends AbstractPureComponent2<Popover2Props<T>, IPopov
             className: classNames(className, Classes.POPOVER2_TARGET, {
                 [Classes.POPOVER2_OPEN]: isOpen,
                 // this class is mainly useful for button targets
-                [CoreClasses.ACTIVE]: !isControlled && isOpen && !isHoverInteractionKind,
+                [CoreClasses.ACTIVE]: isOpen && !isControlled && !isHoverInteractionKind,
             }),
             ref,
             ...(targetEventHandlers as unknown as T),
         };
 
+        const targetModifierClasses = {
+            // this class is mainly useful for Blueprint <Button> targets; we should only apply it for
+            // uncontrolled popovers when they are opened by a user interaction
+            [CoreClasses.ACTIVE]: isOpen && !isControlled && !isHoverInteractionKind,
+            // similarly, this class is mainly useful for targets like <Button>, <InputGroup>, etc.
+            [CoreClasses.FILL]: fill,
+        };
+
         let target: JSX.Element | undefined;
 
         if (renderTarget !== undefined) {
             target = renderTarget({
                 ...targetProps,
+                className: classNames(targetProps.className, targetModifierClasses),
```

Comment:
since we apply the fill class to the child target when renderTarget isn't defined, I figured we should apply this class to the custom target as well